### PR TITLE
Add access token to header

### PR DESCRIPTION
**Pull request by charithag**

### CODE REVIEW

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/api/operation-api.jag [hunk 1]
```diff
@@ -40,6 +40,7 @@ if (!user) {
         var deviceId = request.getParameter("deviceId");
         var monitor_operations = operationModule.getMonitorOperations(deviceType);
```
Comment (harshanL) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/api/operation-api.jag:41]: Use camel-case.

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/operation.js [hunk 2]
```diff
@@ -25,7 +25,18 @@ var operationModule = function () {
     var publicMethods = {};
     var privateMethods = {};
 
-    privateMethods.getOperationsFromFeatures = function (deviceType, type) {
+    /**
+     * This method reads the token from the Token client and return the access token.
+     * If the token pair s not set in the session this will send a redirect to the login page.
+     */
+    function getAccessToken(deviceType, owner, deviceId) {
+        var TokenClient = Package.org.wso2.carbon.device.mgt.iot.apimgt.TokenClient;
```
Comment (harshanL) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/operation.js]: Use camel-case appropriately.

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/operation.js [hunk 3]
```diff
@@ -25,7 +25,18 @@ var operationModule = function () {
...
+        var accessTokenClient = new TokenClient(deviceType);
+        var accessTokenInfo = accessTokenClient.getAccessToken(owner, deviceId);
+        return accessTokenInfo.getAccess_token();
```
Comment (harshanL) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/operation.js:36]: Use camel-case correctly.

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/operation.js [hunk 4]
```diff
@@ -75,15 +86,15 @@ var operationModule = function () {
     publicMethods.handlePOSTOperation = function (deviceType, operation, deviceId, params) {
         var user = session.get(constants.USER_SESSION_KEY);
         var endPoint = devicemgtProps["httpsURL"] + '/' + deviceType + "/controller/" + operation;
-        var header = '{"owner":"' + user.username + '","deviceId":"' + deviceId + '","protocol":"mqtt", "sessionId":"' + session.getId() + '"}';
+        var header = '{"owner":"' + user.username + '","deviceId":"' + deviceId + '","protocol":"mqtt", "sessionId":"' + session.getId() + '", "' + constants.AUTHORIZATION_HEADER + '":"' + constants.BEARER_PREFIX + getAccessToken(deviceType, user.username, deviceId) + '"}';
+        log.warn("header: " + header);
```
Comment (harshanL) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/operation.js:90]: Why does this have a warn level log? Is it necessary/

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/api/operation-api.jag (hunk 1 above)
Comment (charithag) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/api/operation-api.jag:41]: Acknowledged & Fixed with PR https://github.com/wso2/carbon-device-mgt/pull/213

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/operation.js (hunk 3 above)
Comment (charithag) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/operation.js:36]: This method is implemented in another java class.

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/operation.js (hunk 4 above)
Comment (charithag) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/operation.js:90]: Fixed with commit https://github.com/wso2/carbon-device-mgt/commit/e537e2d841ca3c7946cd34e9da96f261e0ac94e1

**Pull request merged**